x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/56_search_application_search_with_apikey.yml

```diff
@@ -0,0 +1,299 @@
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field2
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc1" }
+
+# Query Search Application overriding all parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: value3
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc2" }
+
+# Query Search Application with list of parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application-with-list
+        body:
+          params:
+            query_string: value3
+            text_fields:
+              - name: field1
+                boost: 1
+              - name: field2
+                boost: 2
+              - name: field3
+                boost: 3
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc2" }
+
+# Query Search Application with invalid parameter validation:
+  - do:
+      catch: "bad_request"
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: 35
+
+# Query Search Application without required parameter:
+  - do:
+      catch: "bad_request"
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_value: test
+
+# Query Search Application - not found:
+  - do:
+      catch: forbidden
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: nonexisting-test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: value3
+
+# Get Analytics Collection should be rejected due to a workflow restriction
+  - do:
+      catch: forbidden
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.get_behavioral_analytics:
+        name:
+  - match: { status: 403 }
+  - match: { error.type: security_exception }
+  - match: { error.root_cause.0.type: role_restriction_exception }
+  - match: { error.root_cause.0.reason: "access restricted by workflow" }
+
+# Remove Test API Key:
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      security.invalidate_api_key:
+        body: >
+          {
+            "ids": [ "${api_key_id}" ]
+          }
+  - length: { "invalidated_api_keys": 1 }
```

Comment:
Nit: If we are creating API key in the begining and removing API key at the end, it feels that they should be move to the setup and tear down sections. Otherwise removal will not happen when test fails. But I think in this case, the removal is not something necessary. So we can also just drop this part.

Reply:
Initially, I've tried adding it to `setup` section, but I could not use stashed `api_key_encoded` value outside of setup. I get following error:

```
Caused by: java.lang.IllegalArgumentException: stashed value not found for key [api_key_encoded]
```

Not sure if there is a way to stash API key's secret globally in YAML tests?

Reply:
Hmm good question. It might not be possible or at least I don't remember such an example. We can dig into it later if necessary. But it does not really matter here since whether it is removed does not impact anything. Thanks for explaining.

Reply:
Agreed. I've removed the API key invalidation.